File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-060211-45590/analysis/acceptance-project/src/main/java/customerimporter/DNI.java
package customerimporter;

import java.util.function.Supplier;

public class DNI extends Identification {

    private static final String TYPE = "D";

    // instance creation

    public static boolean canHandle(String aType) {
        return TYPE.equals(aType);
    }

    // error messages

    public static String invalidDNINumberErrorDescription() {
        return "Invalid DNI number";
    }

    // initialization

    public DNI(String aNumber) {
        super(aNumber);
    }

    // accessing

    public String type() {
        return TYPE;
    }

    // testing

    public boolean isDNI() {
        return true;
    }

    public boolean isCUIT() {
        return false;
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return Integer.parseInt(number);
    }

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    // validation

    protected void assertIsValid() {
        assertThat(Characters.areAllDigits(number));
        int dniNumber = Integer.parseInt(number);
        assertThat(dniNumber >= 1 && dniNumber <= 99999999);
    }

    protected String invalidNumberErrorDescription() {
        return invalidDNINumberErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213435-24827/workdir/src/customerimporter/dni.py
from customerimporter.identification import Identification


class Dni(Identification):

    # instance creation

    @classmethod
    def type_code(cls):
        return "D"

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid DNI number"

    # validation

    def assert_is_valid(self):
        self._assert_that(self._are_all_digits(self._value))
        self._assert_that(self.dni_number() >= 1 and self.dni_number() <= 99999999)

    # dni

    def dni_number(self):
        return int(self._value)

    def dni_number_if_none(self, a_none_block):
        return self.dni_number()

    def has_dni_as_identification(self):
        return True
